CMakeLists.txt

```diff
@@ -69,7 +72,11 @@ install(
     TARGETS headers
     EXPORT ${PROJECT_NAME}-targets)
 
+# set the loader target name for linking with other components
+set(TARGET_LOADER_NAME ur_loader)
```

Comment:
Is there really a need to set a variable for this target name?

I think it would be clearer to use the target name explicitly but provide an option to the user to change the name of the shared object via target property.

When adding support for `FetchContent`/`find_package` it became clear that its necessary to have internal/external target names. For example, the internal name for the header only library is `headers` while the external name is `unified-runtime::headers`.

Mu idea would be to follow that pattern for the loader library target as well, with the internal name `loader` and external name `unified-runtime::loader`.

Reply:
It occurs to me this is the kind of thing that should be written down somewhere for future reference. I don't think making this change is really a blocker, I'd be happy to address these changes and document why in a separate PR once this is merged.

Reply:
I removed this variable and added the loader to the unified-runtime namespace.

Reply:
Done. I wanted to remove the `ur_` prefix, but that also changed the library name to `libloader.so`. It turns out there's a `LIBRARY_OUTPUT_NAME` property that can be used to change it back to what we need it to be.